aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,753 @@
+				ValidateFunc: validateArn,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"webserver_access_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  mwaa.WebserverAccessModePrivateOnly,
+				ValidateFunc: validation.StringInSlice([]string{
+					mwaa.WebserverAccessModePrivateOnly,
+					mwaa.WebserverAccessModePublicOnly,
+				}, false),
+			},
+			"webserver_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"weekly_maintenance_window_start": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsMwaaEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mwaaconn
+
+	input := mwaa.CreateEnvironmentInput{
+		DagS3Path:            aws.String(d.Get("dag_s3_path").(string)),
+		ExecutionRoleArn:     aws.String(d.Get("execution_role_arn").(string)),
+		Name:                 aws.String(d.Get("name").(string)),
+		NetworkConfiguration: expandMwaaEnvironmentNetworkConfigurationCreate(d.Get("network_configuration").([]interface{})),
+		SourceBucketArn:      aws.String(d.Get("source_bucket_arn").(string)),
+	}
+
+	if v, ok := d.GetOk("airflow_configuration_options"); ok {
+		input.AirflowConfigurationOptions = stringMapToPointers(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("airflow_version"); ok {
+		input.AirflowVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("environment_class"); ok {
+		input.EnvironmentClass = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("kms_key"); ok {
+		input.KmsKey = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("logging_configuration"); ok {
+		input.LoggingConfiguration = expandMwaaEnvironmentLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("max_workers"); ok {
+		input.MaxWorkers = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("plugins_s3_object_version"); ok {
+		input.PluginsS3ObjectVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("plugins_s3_path"); ok {
+		input.PluginsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("requirements_s3_object_version"); ok {
+		input.RequirementsS3ObjectVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("requirements_s3_path"); ok {
+		input.RequirementsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("webserver_access_mode"); ok {
+		input.WebserverAccessMode = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("weekly_maintenance_window_start"); ok {
+		input.WeeklyMaintenanceWindowStart = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		input.Tags = keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().MwaaTags()
+	}
+
+	log.Printf("[INFO] Creating MWAA Environment: %s", input)
+	out, err := conn.CreateEnvironment(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(aws.StringValue(input.Name))
+	d.Set("arn", out.Arn)
```

Comment:
No need to set `arn` here as it will be set in `resourceAwsMwaaEnvironmentRead` called before successful return.